plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPartitioningHandle.java

```diff
@@ -92,16 +113,14 @@ private static boolean buildDataPaths(Set<Integer> partitionFieldIds, Types.Stru
             currentPaths.addLast(fieldOrdinal);
             org.apache.iceberg.types.Type type = field.type();
             if (type instanceof Types.StructType nestedStruct) {
-                hasPartitionFields = buildDataPaths(partitionFieldIds, nestedStruct, currentPaths, dataPaths) || hasPartitionFields;
+                buildDataPaths(partitionFieldIds, nestedStruct, currentPaths, dataPaths);
             }
             // Map and List types are not supported in partitioning
             if (type.isPrimitiveType() && partitionFieldIds.contains(fieldId)) {
```

Comment:
should this be `else if`?  I assume structure is mutually exclusive of these cases. If so, I think it makes more sense to be explicit.  If not, we should explain what the consequences of that is.